apps/synapse-oauth-signin/src/mocks/handlers.ts

```diff
@@ -20,115 +28,124 @@ export const URL_ENCODED_ACCESS_CODE_PROVIDED_BY_SERVER =
+      ),
+  ),
+  http.get<never, never, TwoFactorAuthStatus>(
+    'https://repo-prod.prod.sagebase.org/auth/v1/2fa',
+    () =>
+      HttpResponse.json(
+        {
+          status: 'ENABLED',
+        },
+        { status: 200 },
+      ),
   ),
 
-  rest.get(
+  http.get('https://repo-prod.prod.sagebase.org/repo/v1/userProfile', () => {
+    return HttpResponse.json({}, { status: 200 })
+  }),
+
+  http.get(
     'https://repo-prod.prod.sagebase.org/auth/v1/oauth2/client/:id',
-    (req, res, ctx) => {
-      return res(ctx.status(200), ctx.json(mockOauthClient))
+    () => {
+      return HttpResponse.json(mockOauthClient, { status: 200 })
     },
   ),
 
-  rest.post(
+  http.post(
     'https://repo-prod.prod.sagebase.org/auth/v1/oauth2/description',
-    (req, res, ctx) => {
-      return res(
-        ctx.status(200),
-        ctx.json({
+    () =>
+      HttpResponse.json(
+        {
           clientId: '1234',
           redirect_uri: 'https://some-redirect-uri.abc/redirect',
           scope: [
             'To see your Synapse user ID, which can be used to access your public profile',
           ],
-        }),
-      )
-    },
+        },
+        { status: 200 },
+      ),
   ),
 
-  rest.post(
-    'https://repo-prod.prod.sagebase.org/auth/v1/login2',
-    (req, res, ctx) => {
-      return res(
-        ctx.status(200),
-        ctx.json({
-          accessToken: 'someToken',
-          acceptsTermsOfUse: true,
-        }),
-      )
-    },
+  http.post('https://repo-prod.prod.sagebase.org/auth/v1/login2', () =>
+    HttpResponse.json(
+      { accessToken: 'someToken', acceptsTermsOfUse: true },
+      { status: 200 },
+    ),
   ),
 
-  rest.post(
+  http.post(
     'https://repo-prod.prod.sagebase.org/auth/v1/oauth2/consentcheck',
-    async (req, res, ctx) => {
-      return res(
-        ctx.status(200),
-        ctx.json({
-          granted: hasConsented,
-        }),
-      )
-    },
+    () => HttpResponse.json({ granted: hasConsented }, { status: 200 }),
   ),
 
-  rest.post(
+  http.post(
     'https://repo-prod.prod.sagebase.org/auth/v1/oauth2/consent',
-    (req, res, ctx) => {
+    () => {
       hasConsented = true
       mockPostConsentFn()
-      return res(
-        ctx.status(200),
-        ctx.json({
-          access_code: ACCESS_CODE_PROVIDED_BY_SERVER,
-        }),
+      return HttpResponse.json(
+        { access_code: ACCESS_CODE_PROVIDED_BY_SERVER },
+        { status: 200 },
       )
     },
   ),
 
-  rest.delete(
+  http.delete(
     'https://repo-prod.prod.sagebase.org/auth/v1/sessionAccessToken',
-    (req, res, ctx) => {
-      return res(ctx.status(200), ctx.json(''))
+    () => {
+      return HttpResponse.json('', { status: 200 })
     },
   ),
 ]
 
 export function getOAuth2DescriptionWithUnverifiedClientHandler() {
-  return rest.post(
+  return http.post<
+    never,
+    OIDCAuthorizationRequest,
+    SynapseApiResponse<OIDCAuthorizationRequestDescription, OAuthErrorResponse>
```

Comment:
In many cases, add generic arguments to ensure our handlers are accepting/returning the correct data.